nwb_conversion_tools/datainterfaces/spikeglxdatainterface.py

```diff
@@ -1,15 +1,58 @@
 """Authors: Cody Baker and Ben Dichter."""
+from pathlib import Path
 from spikeextractors import SpikeGLXRecordingExtractor
 
 from ..baserecordingextractorinterface import BaseRecordingExtractorInterface
 
 
 class SpikeGLXRecordingInterface(BaseRecordingExtractorInterface):
-    """Primary data interface class for converting a NeuroscopeRecordingExtractor."""
+    """Primary data interface class for converting the SpikeGLX format."""
 
     RX = SpikeGLXRecordingExtractor
 
-    def get_metadata():
-        """Retrieve Ecephys metadata specific to the SpikeGLX format."""
-        # TODO
-        raise NotImplementedError
+    def get_metadata(self):
+        """Auto-populate as much metadata as possible from the SpikeGLX format."""
+        file_path = Path(self.input_args['file_path'])
+        session_id = file_path.parent.stem
+
+        n_shanks = int(self.recording_extractor._meta['snsShankMap'][1])
+        if n_shanks > 1:
+            raise NotImplementedError("SpikeGLX metadata for more than a single shank is not yet supported.")
+
+        channels = self.recording_extractor.get_channel_ids()
+        shank_electrode_number = channels
+        shank_group_name = ["Shank1" for x in channels]
+
+        re_metadata = dict(
+            Ecephys=dict(
+                Device=[
+                    dict(
+                        description=session_id + '.ap.meta'
+                    )
+                ],
+                ElectrodeGroup=[
+                    dict(
+                        name='Shank1',
+                        description="Shank1 electrodes."
+                    )
+                    for n in range(n_shanks)
+                ],
+                Electrodes=[
+                    dict(
+                        name='shank_electrode_number',
+                        description="0-indexed channel within a shank.",
+                        data=shank_electrode_number
+                    ),
+                    dict(
+                        name='group_name',
+                        description="The name of the ElectrodeGroup this electrode is a part of.",
+                        data=shank_group_name
+                    )
+                ],
+                ElectricalSeries=dict(
+                    name='ElectricalSeries',
+                    description="Raw acquisition traces."
```

Comment:
I believe spikeGLX records two different datasets, ap, which is high-passed, and lf, which is low-passed. Could you reflect this in the metadata structure with descriptions?

Reply:
Updated the Series and device (since there are two different meta files) descriptions as well as docstrings; I'm assuming the lf would use the same electrodes, so haven't modified those.